test/with_dummyserver/test_https.py

```diff
@@ -974,11 +975,18 @@ def test_can_validate_san(self, san_server: ServerConfig) -> None:
             assert r.status == 200
 
     def test_common_name_without_san_fails(self, no_san_server: ServerConfig) -> None:
```

Comment:
This test should continue to pass without the modifications IMO, since our default ssl_context should have `hostname_checks_common_name = False` set.

Reply:
I've changed the code to pass this test.
Now, this test fails on `pypy` because [_is_bpo_43522_fixed](https://github.com/urllib3/urllib3/blob/49848a178c1ad108d311bdf904bd63aa3fd593e0/src/urllib3/util/ssl_.py#L59:L68) just works for `cpython`.
Shall we include `pypy` in this function?

Reply:
Hmm, yeah can we figure out which versions of PyPy have that fix and go off of that? I guess I'd prefer being conservative and returning `False` from that function anywhere where we're not **certain** that setting `SSLContext.hostname_checks_common_name = False` works.

Can we somehow create a unit test for our function to see that the value of `urllib3.util.ssl_.HAS_NEVER_CHECK_COMMON_NAME` is the same as what `SSLContext.hostname_checks_common_name`'s behavior is?

This might be a thing we can do outside this PR, but for now let's return `False` for PyPy and lean on our own hostname checking implementation.

Reply:
For what it's worth, I've intentionally checked for CPython only in `_is_bpo_43522_fixed`, because on PyPy, `ssl.HAS_NEVER_CHECK_COMMON_NAME` is supposed to always be reliable when present, see https://github.com/urllib3/urllib3/issues/2191#issuecomment-902618341 and https://foss.heptapod.net/pypy/pypy/-/issues/3539.

Reply:
Okay so we can leave the implementation as-is, thanks Quentin!